playground/flappy-bird.ts

```diff
@@ -0,0 +1,798 @@
+  private _myScore = 0;
+
+  onAwake() {
+    // Init spriteArray
+    const spriteArray = this._spriteArray;
+    for (var i = 0; i < 10; i++) {
+      spriteArray.push(new Sprite(engine, gameResArray[5], new Rect(i * 0.1, 0, 0.1, 1)));
+    }
+
+    engine.on(GameEvent.addScore, () => {
+      ++this._myScore;
+      this.showScore("" + this._myScore);
+    });
+
+    // Control the performance of the score according to the change of the game state
+    engine.on(GameEvent.stateChange, (gameState: EnumGameState) => {
+      switch (gameState) {
+        case EnumGameState.Idel:
+          this.entity.isActive = false;
+          break;
+        case EnumGameState.Start:
+          this._myScore = 0;
+          this.entity.isActive = true;
+          this.showScore("0");
+          break;
+        case EnumGameState.End:
+          break;
+      }
+    });
+  }
+
+  private showScore(scoreNumStr: string) {
+    const scoreLen = scoreNumStr.length;
+    const { entity, _numInv: inv, _scoreEntitys: scoreEntitys, _spriteArray: spriteArray } = this;
+    var nowEntityLen = scoreEntitys.length;
+    let scoreEntity: NumberEntity;
+    // If the entity is not enough, new one
+    if (scoreLen > nowEntityLen) {
+      for (let i = nowEntityLen; i < scoreLen; i++) {
+        scoreEntity = new NumberEntity(engine, "num" + i);
+        scoreEntitys.push(scoreEntity);
+        entity.addChild(scoreEntity);
+      }
+    }
+
+    // At the moment nowEntityLen >= scoreLen
+    nowEntityLen = scoreEntitys.length;
+    const startX = ((1 - scoreLen) * inv) / 2;
+    for (let i = 0; i < nowEntityLen; i++) {
+      scoreEntity = scoreEntitys[i];
+      if (i >= scoreLen) {
+        scoreEntity.isActive = false;
+      } else {
+        scoreEntity.isActive = true;
+        scoreEntity.transform.setPosition(startX + i * inv, 0, 0);
+        scoreEntity.showNumber(spriteArray[parseInt(scoreNumStr[i])]);
+      }
+    }
+  }
+}
+
+class ScriptGround extends Script {
+  // Swap two pieces of ground to achieve constant movement
+  private leftGround: Entity;
+  private rightGround: Entity;
+
+  onAwake() {
+    const entity = this.entity;
+    this.leftGround = entity.findByName("ground1");
+    this.rightGround = entity.findByName("ground2");
+    // Control the performance of the ground according to the change of the game state
+    engine.on(GameEvent.stateChange, (gameState: EnumGameState) => {
+      switch (gameState) {
+        case EnumGameState.Idel:
+        case EnumGameState.Start:
+          this.enabled = true;
+          break;
+        case EnumGameState.End:
+          this.enabled = false;
+          break;
+        default:
+          break;
+      }
+    });
+
+    // When checkHit is monitored, check the collision between the ground and the bird
+    engine.on(GameEvent.checkHit, (birdY) => {
+      birdY < groundY && engine.dispatch(GameEvent.gameOver);
+    });
+  }
+
+  onUpdate(deltaTime: number) {
```

Comment:
Use `tileOffset` feature of material is better.

Reply:
done.